### PR TITLE
Kafka stabiltiy test

### PR DESCRIPTION
**Pull request by howardjohn**

### CODE REVIEW

**Comment by GregHanson** [comment]
@howardjohn will this test eventually change as kafka support is merged into envoy?  Several PR's have started going in the past few days, such as https://github.com/envoyproxy/envoy/pull/4950

**Comment by howardjohn** [comment]
@GregHanson I had no idea about that, thanks. We will need to add support to istio as well, right? Then just change port names to `kafka-`?

**Comment by GregHanson** [comment]
yes, eventually we will have to pull this into istio - kafka is something that comes up pretty frequently with istio in my experience.  @cmluciano might be able to provide some more insight into this from an envoy perspective.  As it stands now, the codec has been merged into envoy but not the filter?

**Comment by GregHanson** [comment]
but basically it sounds like envoy still has some work to do on its end, so the test as it stands now ( with basic ServiceEntry with Resolution type NONE) is definitely a good test to have for the time being

**Comment by howardjohn** [comment]
The ServiceEntry won't change when we get the envoy filter working though , it is needed because Istio won't create the entries for every instance of the headless service so they cannot talk to each other without it, unless I am doing it wrong

File: perf/docker/kafka/client.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
+import os
```
Comment (GregHanson) [inline perf/docker/kafka/client.py]: Please addd license to the top of this file similar to the one in our helloworld app:
https://github.com/istio/istio/blob/master/samples/helloworld/src/app.py#L1

**Comment by howardjohn** [comment]
@mandarjog @utka can you give owners approval

**Comment by hansel801** [comment]
Hi, have you tried Kafka with sidecar + mTLS enabled? Wondering if this works...

**Pull request closed without merging**